### PR TITLE
[BotBuilder-AI] Make BotBuilder AI web compatible

### PR DESCRIPTION
**Pull request by GasparAcevedoZainSouthworks**
Fixes issue #620.

Must be merged with PR #909.

## Description
Remove or replace NodeJS dependent libraries, and make the **BotBuilder-AI** library fully web-compatible.

## Changes
- Replace `request-promise-native` with `node-fetch`
- `@azure/cognitiveservices-luis-runtime` was fixed in pr # 844
- `html-entities` should be removed to keep parity with the .NET version. We will be creating a Pull Request with this change and more detailed information regarding this decision.
- Remove `request` since it was unused.

## Details
We searched through the library looking for any incompatible dependencies to remove.

 The libraries already identified by the original issue report as ones to be removed were:
•	**request-promise-native** - Suggested to be replaced by node-fetch
•	**azure-cognitiveservices-luis-runtime** - Suggested to be replaced by @azure/cognitiveservices-luis-runtime
•	**html-entities**
 
_request-promise-native_ was removed and replaced by node-fetch
_html-entities_ was straight out removed and its removal documented at `PR 909`
_azure-cognitiveservices-luis-runtime_ had been already removed by a previous commit at `e6016d4226656378aa1a370e022a8755e25e3609`
 
Furthermore, we researched the compatibility of the rest of the dependencies in **BotBuilder-AI** looking for some more libraries that could break compatibility with web deployment. The remaining dependencies then were:
 
image
 
## Testing
We used the dependencies' documentation, when specified, to check which libraries were ready for web. 
For a better, practical way, to test the web-compatibility of the library we set to deploy a bot to the web and make calls to the BotBuilder-AI. That would both allow us to test any changes we would make as well as the current state of compatibility of the library.
 
Looking at the index.ts file, several of **BotBuilder-AI** exports are internally-used structures; and aside from those, the two major exports are **LuisRecognizer** and **QnAMaker**. We made test calls to both in order for the bot to query these services.
 
image
 
As our bot we chose to use the browser **echobot project** (since it's a bot that can be deployed to web and run on a browser) and modify part of its code to make calls to **BotBuilder-AI**. We made some dummy calls at the beginning of the processActivity method, meaning the code would be triggered any time a conversation state or a message update arrived.
 
image
 
To check the response, we added the result to the existing echobot response

image
 
We got correct responses both querying the QnA service 
 
image

and the LUIS service:
 
image
 
Since the library is apparently already correctly working when deployed online (after the removals specified at the top of this post) we only ended up removing _**request**_ as a dependency since we found it was unused. The rest of the dependencies performed correctly when calling the library from a web bot.

### CODE REVIEW

**Comment by cleemullins** [comment]
@stevengum Can you review + merge this in today?

File: libraries/botbuilder-ai/src/qnaMaker.ts [hunk 1]
```diff
@@ -458,18 +458,19 @@ export class QnAMaker implements QnAMakerTelemetryClient {
 
         this.validateOptions(queryOptions);
 
-        const qnaResult: any = await request({
-            url: url,
+        const qnaResult: any = await request( url, {
```
Comment (stevengum) [inline libraries/botbuilder-ai/src/qnaMaker.ts]: extra space after `(`

File: libraries/botbuilder-ai/src/qnaMaker.ts [hunk 2]
```diff
@@ -458,18 +458,19 @@ export class QnAMaker implements QnAMakerTelemetryClient {
...
             method: 'POST',
             headers: headers,
             timeout: queryOptions.timeout,
-            json: {
+            body: JSON.stringify( {
```
Comment (stevengum) [inline libraries/botbuilder-ai/src/qnaMaker.ts]: Another extra space after `(`

**Comment by stevengum** [comment]
@cleemullins there's some minor code cleanup and the package-lock.json for the library needs to be updated, but after those changes are in this one is good to go.

@GasparAcevedoZainSouthworks

**Pull request merged**